Fix set_static_all, always display requestmode and refactor _change_state

alexvasiu opened the pull request:
* modules/nilrt_ip.py: Display requestmode even if service is offline
This will keep parity between older and newer nilrt distro.

* modules/nilrt_ip.py: Split _change_state function
Split the logic of _change_state function for older and newer disto.

* modules/nilrt_ip.py: Fix set_static_all
If there are no nameservers specified, we shouldn't set any property
for nameservers configuration to the service.

Commit:
modules/nilrt_ip.py: Fix set_static_all

If there are no nameservers specified, we shouldn't set any property
for nameservers configuration to the service.

Signed-off-by: Alexandru Vasiu <alexandru.vasiu@ni.com>

Commit:
modules/nilrt_ip.py: Split _change_state function

Split the logic of change_state function for older and
newer disto.

Signed-off-by: Alexandru Vasiu <alexandru.vasiu@ni.com>

Commit:
modules/nilrt_ip.py: Display requestmode even if service is offline

This will keep parity between older and newer nilrt distro.

Signed-off-by: Alexandru Vasiu <alexandru.vasiu@ni.com>

rares-pop commented on salt/modules/nilrt_ip.py:
```diff
@@ -510,9 +513,33 @@ def get_interfaces_details():
 
 
-def _change_state(interface, new_state):
+def _change_state_older(interface, new_state):
```
we should go with 'legacy' instead of 'older', as it is in other modules.

I'd rename this to something like: _change_state_legacy()

rares-pop commented on salt/modules/nilrt_ip.py:
```diff
@@ -510,9 +513,33 @@ def get_interfaces_details():
+    return True
+
+
+def _change_state_newer(interface, new_state):
```
if the other is _change_state_legacy(), this can be just _change_state()

alexvasiu replied:
Fixed

rares-pop reviewed (changes requested):
we'd need some tests too for those

rares-pop commented on salt/modules/nilrt_ip.py:
```diff
@@ -522,17 +549,7 @@ def _change_state(interface, new_state):
-                                                                       interface, out['stderr'])
-                raise salt.exceptions.CommandExecutionError(msg)
-        return True
+        return _change_state_legacy(interface, new_state)
```
Please move this at the top of the function, so it's easier to follow what happens for legacy systems, vs the rest. Right now is in the middle of the function.

The code should look like: 

```
if legacy:
 return _change_state_legacy()
do the non-legacy behavior
```

rares-pop replied:
Nevermind.. my bad!

rares-pop reviewed (approved):
Looks good!

alexvasiu commented:
@rallytime Can you take a look?

cachedout commented on salt/modules/nilrt_ip.py:
```diff
@@ -216,12 +216,15 @@ def _get_service_info(service):
+    data['ipv4']['supportedrequestmodes'] = [
+        'static',
+        'dhcp_linklocal',
+        'disabled'
```
Was switching the order of these intentional?

alexvasiu replied:
Nope. I can sort them, if it's better

Commit:
modules/nilrt_ip.py: Parity between CG and NG for all request modes

All request possible request modes for NG are:
    - dhcp_linklocal
    - disabled
    - static

Signed-off-by: Alexandru Vasiu <alexandru.vasiu@ni.com>
